src/app/zap-templates/templates/app/helper.js

```diff
@@ -194,30 +194,41 @@ function chip_endpoint_generated_functions()
 
 function chip_endpoint_generated_commands_list(options)
 {
-  let ret = [];
+  let ret   = [];
+  let index = 0;
   this.clusterList.forEach((c) => {
     let acceptedCommands  = [];
     let generatedCommands = [];
 
     c.commands.forEach((cmd) => {
       if (cmd.mask.includes('incoming_server')) {
         acceptedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
-      }
-      if (cmd.mask.includes('incoming_client')) {
-        generatedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+        console.log(cmd);
+        if (cmd.responseName) {
```

Comment:
This seems reasonable, but what was going wrong before?  Why was `SetRegulatoryConfigResponse` running into problems, while `ArmFailSafeResponse` (which looks to be configured identically) was not?  Is it because `SetRegulatoryConfigResponse` is (incorrectly) marked optional in the XML?

Reply:
There are some issues in ZAP the zap files.

We have four flags for a command -- (incoming outgoing) x (server client)

However, in Matter, all commands from server to client are response to another from client to server (as indicated by the spec 7.13.3)

Thus some clusters in the ZAP file are mistakenly marked as "client" when they should be server, in this case, some commands have wrong flag -- both as incoming and outgoing commands.

By using the same algorithm as described in spec, we should have a more consistent result.

Reply:
Filed https://github.com/project-chip/zap/issues/481 to ZAP